addons/common/functions/fnc_adjustMagazineAmmo.sqf

```diff
@@ -0,0 +1,63 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Katalam, Blue, Brett Mayson
+ * Handle adjusting a magazine's ammo
+ *
+ * Arguments:
+ * 0: Vehicle or Unit <OBJECT>
+ * 1: Item <STRING>
+ * 2: Count <NUMBER>
+ *
+ * Return Value:
+ * Empty mag <BOOLEAN> - true if the magazine is empty after the ammo is consumed
+ *
+ * Example:
+ * [player, "30Rnd_556x45_Stanag", 1] call ace_commone_fnc_adjustMagazineAmmo;
+ *
+ * Public: No
+ */
+
+params ["_unit", "_item", ["_count", -1]];
+
+private _lowestCount = 999;
+private _lowestIndex = 0;
+
+if (_unit isKindOf "CAManBase") then {
+    private _magazines = [];
+    {
+        _x params ["_class", "_ammo", "_loaded", "", "_location"];
+        if (_class == _item && !_loaded) then {
+            _magazines pushBack _x;
+            if (_ammo < _lowestCount) then {
+                _lowestCount = _ammo;
+                _lowestIndex = count _magazines - 1;
+            };
+        };
+    } forEach (magazinesAmmoFull _unit);
```

Comment:
Can't we use `_magazines = magazinesAmmo select {(_x select 0) == _item}` and then `_magazines sort false` instead?